internal/pkg/coordinator/monitor.go

```diff
@@ -393,21 +396,24 @@ func (m *monitorT) getIPs() ([]string, error) {
 }
 
 func (m *monitorT) rescheduleUnenroller(ctx context.Context, pt *policyT, p *model.Policy) {
-	l := m.log.With().Str(dl.FieldPolicyId, pt.id).Logger()
+	u := uuid.Must(uuid.NewV4())
+	l := m.log.With().Str(dl.FieldPolicyId, pt.id).Str("unenroller_uuid", u.String()).Logger()
 	unenrollTimeout := time.Duration(p.UnenrollTimeout) * time.Second
 	if unenrollTimeout != pt.unenrollTimeout {
 		// unenroll timeout changed
-		if pt.unenrollCanceller != nil {
-			pt.unenrollCanceller()
-			pt.unenrollCanceller = nil
+		if c, ok := m.policiesCanceller[pt.id]; ok {
+			l.Debug().Msg("cancelling unenrollment monitor")
```

Comment:
It would also be created to add more detail on why it was cancelled, say like because timeout changed or because timeout was unset.